### ISSUE TITLE
CxFlow closing Github issues before scan is complete

### ISSUE THREAD

**Issue by aqiu95** [checkmarx-ltd/cx-flow#1008]
### Description

We are seeing that CxFlow is occasionally closing Github issues before a scan is complete, and then reopening those issues once the scan is complete, which leads to CxFlow repeatedly opening and closing Github issues, leaving developer teams confused as to the status of a finding.

### Expected Behavior

CxFlow should only close Github issues after the scan is complete and those issues are no longer present.

### Actual Behavior

CxFlow closes Github issues minutes after a scan is submitted (while it is still in source pulling & deployment), and then will later reopen an issue once the scan is complete.

### Reproduction

This seems to happen only on some repos, but will consistently happen to the same repos. Some example snippets of CxFlow logs:

```
{"time": "2022-04-27 03:04:50.075", "level": "INFO", "correlationId": "SUPLaC9j", "env": "prod", "region": "us-east-1", "source": "com.checkmarx.sdk.service.CxService:1810", "message": "Scan created with Id 1936710 for project Id 41279"}
{"time": "2022-04-27 03:04:50.080", "level": "INFO", "correlationId": "SUPLaC9j", "env": "prod", "region": "us-east-1", "source": "com.checkmarx.flow.service.ResultsService:218", "message": "Issue tracking is custom bean implementation"}
{"time": "2022-04-27 03:04:50.080", "level": "INFO", "correlationId": "SUPLaC9j", "env": "prod", "region": "us-east-1", "source": "com.checkmarx.flow.custom.GitHubIssueTracker:54", "message": "======== Initializing GitHub processing ========"}
{"time": "2022-04-27 03:04:50.080", "level": "INFO", "correlationId": "SUPLaC9j", "env": "prod", "region": "us-east-1", "source": "com.checkmarx.flow.service.CodeBashingService:48", "message": "not using CodeBashing lessons integration - one or more of the mandatory properties is missing"}
{"time": "2022-04-27 03:04:50.080", "level": "INFO", "correlationId": "SUPLaC9j", "env": "prod", "region": "us-east-1", "source": "com.checkmarx.flow.service.IssueService:105", "message": "Processing Issues with custom bean GitHub"}
{"time": "2022-04-27 03:04:50.080", "level": "INFO", "correlationId": "SUPLaC9j", "env": "prod", "region": "us-east-1", "source": "com.checkmarx.flow.custom.GitHubIssueTracker:80", "message": "Executing getIssues GitHub API call: https://github.ourdomain.com/api/v3/repos/orgname/reponame/issues?state=all&per_page=100"}
{"time": "2022-04-27 03:07:38.660", "level": "INFO", "correlationId": "SUPLaC9j", "env": "prod", "region": "us-east-1", "source": "com.checkmarx.flow.custom.GitHubIssueTracker:192", "message": "Executing closeIssue GitHub API call"}
{"time": "2022-04-27 03:07:38.874", "level": "INFO", "correlationId": "SUPLaC9j", "env": "prod", "region": "us-east-1", "source": "com.checkmarx.flow.service.IssueService:181", "message": "Closing issue #6827098 with key CX Unsafe_Use_Of_Target_blank @ path/to/file/file.ts [master]"}
{"time": "2022-04-27 03:07:38.875", "level": "INFO", "correlationId": "SUPLaC9j", "env": "prod", "region": "us-east-1", "source": "com.checkmarx.flow.custom.GitHubIssueTracker:192", "message": "Executing closeIssue GitHub API call"}
```

In this particular case, the scan ID 1936710 for project 41279 has a start time of 4/27/2022 3:58:17 AM and a completion time of 4/27/2022 5:35:22 AM according to the manager, which means these issues were closed before the scan even started. On the next scan, these issues will be reopened.

### Environment Details

CxFlow 1.6.30 running on AWS ECS on RHEL7.

**Comment by Kapombo** [comment]
This is consistent with logic in AbstractVulnerabilityScanner that returns empty results if "enable-post-action-monitor" is enabled: 

Empty results returned would mean anything open would be closed, as shown in logic in FlowService:  

FlowService does not check that "enable-post-action-monitor" is being used, so will execute the issue lifecycle management on an empty set of results. The postback event comes along and re-opens what is still in the scan and adds any new issues. 

Checking for the use of "enable-post-action-monitor" before publishing results in FlowService should fix this issue

**Comment by HussainS12** [comment]
Fixed in PR #1023
